src-core/src/generation/generate.rs

```diff
@@ -350,17 +355,81 @@ pub fn generate(
     // for o in polygonObstacles {
     //     path_builder.sgmt_polygon_obstacle(0, wpt_cnt - 1, o.x, o.y, o.radius);
     // }
-    traj_builder.set_drivetrain(&drivetrain);
-    //Err("".to_string())
-    let result = traj_builder
-        .generate(true, handle)
-        .map_err(ChoreoError::TrajOpt)?;
+    traj_builder.set_bumpers(
+        config.bumper.back + config.bumper.front,
+        config.bumper.left + config.bumper.right,
+    );
+    let samples = match traj_builder.as_any().downcast_mut::<SwervePathBuilder>() {
+        Some(traj_builder) => {
+            let drivetrain = SwerveDrivetrain {
+                mass: config.mass,
+                moi: config.inertia,
+                wheel_radius: config.radius,
+                // rad per sec
+                wheel_max_angular_velocity: config.vmax / config.gearing,
+                wheel_max_torque: config.tmax * config.gearing,
+                modules: config
+                    .modules
+                    .map(|modu: Module<f64>| modu.translation())
+                    .to_vec(),
+            };
+
+            traj_builder.add_progress_callback(swerve_status_callback);
 
-    Ok(postprocess(&result, path, snapshot, counts_vec))
+            traj_builder.set_drivetrain(&drivetrain);
```

Comment:
See if you can split up the generate function, ill do this further in a future pr but this function is getting quite monolithic